### PR TITLE
base-wxwidgets: New bootstrap project.

### PR DESCRIPTION
**Pull request by williamjcm**
I hope I did it correctly. 😅 

Notes: builds on MSYS2/MinGW-w64, with Corrade/Magnum 2019.01 and wxWidgets 3.0.4 from the MSYS2 repos. The 3.1 code path works in another project of mine, so it *should* "just work" too.

Anyway, feedback and reviews are welcome.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+cmake_minimum_required(VERSION 3.1)
+project(MyApplication)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/modules/")
```
Comment (mosra) [inline CMakeLists.txt]: Ah damn, that reminds me I forgot to update the bootstrap repos to do

```cmake
set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/modules/" ${CMAKE_MODULE_PATH})
```

instead.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_EXTENSIONS ON)
```
Comment (mosra) [inline src/CMakeLists.txt]: There *might* be some bad interactions between this and the `-std=` flag added by Corrade with the `CORRADE_CXX_STANDARD` property. I'll need to check.

File: src/MyApplication.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+using namespace Magnum;
+
+class MyApplication: public wxApp {
+    public:
+        bool OnInit();
+};
+
+class MainFrame: public wxFrame {
+    public:
+        MainFrame(int argc, char** argv);
```
Comment (mosra) [inline src/MyApplication.cpp]: `explicit`?

File: src/MyApplication.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    MainFrame *frame = new MainFrame{argc, argv};
+    frame->Show(true);
+    return true;
+}
+
+MainFrame::MainFrame(int argc, char** argv): wxFrame{nullptr, wxID_ANY, "Magnum wxWidgets Application"}, _glContext{NoCreate, argc, argv} {
+    wxBoxSizer* bSizer;
+    bSizer = new wxBoxSizer{wxVERTICAL};
+
+#if (wxMAJOR_VERSION == 3) && (wxMINOR_VERSION >= 1)
```
Comment (mosra) [inline src/MyApplication.cpp]: I'm always indenting the preprocessor. (I can also do the change while merging, no worries.)

File: CMakeLists.txt (hunk 1 above)
Comment (williamjcm) [reply CMakeLists.txt]: Fixed locally, will push soon.

File: src/MyApplication.cpp (hunk 3 above)
Comment (williamjcm) [reply src/MyApplication.cpp]: Added locally. Will push soon.

File: src/MyApplication.cpp (hunk 4 above)
Comment (williamjcm) [reply src/MyApplication.cpp]: Indented locally. Will push soon with the other changes.

File: src/CMakeLists.txt (hunk 2 above)
Comment (williamjcm) [reply src/CMakeLists.txt]: I didn't notice anything weird on my end so far. Both with the bootstrap project and my main project, but maybe I'm just lucky.

Comment (mosra) [reply src/CMakeLists.txt]: For now I'll change this to `CORRADE_CXX_STANDARD` like everywhere else, which kinda works around the need to test anything :D

**Comment by mosra** [comment]
Great, it just built [on the first attempt](https://travis-ci.org/mosra/magnum-bootstrap/jobs/533844384) :tada: Merged as 44ec5f38d7496b5bc957b1b3df61c2b1866e362b, thanks a lot!

**Pull request closed without merging**